src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -158,9 +158,7 @@ public IAfterCallSpecifiedConfiguration CallsBaseMethod()
 
             public IReturnValueConfiguration<TMember> WhenArgumentsMatch(Func<ArgumentCollection, bool> argumentsPredicate)
             {
-                Guard.AgainstNull(argumentsPredicate, nameof(argumentsPredicate));

```

Comment:
`ParentConfiguration.WhenArgumentsMatch` already has the `Guard`, so it would be redundant to check it again